wpilibc/src/main/native/cpp/shuffleboard/SimpleWidget.cpp

```diff
@@ -0,0 +1,45 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "frc/shuffleboard/SimpleWidget.h"
+
+#include "frc/shuffleboard/Shuffleboard.h"
+#include "frc/shuffleboard/ShuffleboardLayout.h"
+#include "frc/shuffleboard/ShuffleboardTab.h"
+
+using namespace frc;
+
+SimpleWidget::SimpleWidget(ShuffleboardContainer& parent, wpi::StringRef title)
+    : ShuffleboardValue(title), ShuffleboardWidget(parent, title) {}
+
+nt::NetworkTableEntry SimpleWidget::GetEntry() {
+  if (!m_entryInitialized) {
+    ForceGenerate();
+    m_entryInitialized = true;
+  }
+  return m_entry;
+}
+
+void SimpleWidget::BuildInto(std::shared_ptr<nt::NetworkTable> parentTable,
+                             std::shared_ptr<nt::NetworkTable> metaTable) {
+  BuildMetadata(metaTable);
+  if (!m_entryInitialized) {
+    m_entry = parentTable->GetEntry(GetTitle());
+    m_entryInitialized = true;
+  }
+}
+
+void SimpleWidget::ForceGenerate() {
+  ShuffleboardContainer* parent = &GetParent();
+
+  while (dynamic_cast<ShuffleboardLayout*>(parent) != nullptr) {
```

Comment:
Let's avoid the dynamic_cast here.  Since we control these classes, we should be able to add a tag value to make this much faster and also avoid the dynamic_cast.